mobile/lib/widgets/profile/follow_from_profile_button.dart

```diff
@@ -0,0 +1,97 @@
+// ABOUTME: Follow button widget for profile page using BLoC pattern.
+// ABOUTME: Uses Page/View pattern - Page creates BLoC, View consumes it.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:openvine/blocs/following/following_bloc.dart';
+import 'package:openvine/providers/app_providers.dart';
+import 'package:openvine/providers/nostr_client_provider.dart';
+import 'package:openvine/theme/vine_theme.dart';
+import 'package:openvine/utils/unified_logger.dart';
+
+/// Page widget that creates the [FollowingBloc] and provides it to the view.
+///
+/// Follows the VGV Page/View pattern:
+/// - Page: Creates and provides BLoC with dependencies
+/// - View: Consumes BLoC state and renders UI
+class FollowFromProfileButton extends ConsumerWidget {
+  const FollowFromProfileButton({super.key, required this.pubkey});
+
+  /// The public key of the profile user to follow/unfollow.
+  final String pubkey;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final followRepository = ref.watch(followRepositoryProvider);
+    final nostrClient = ref.watch(nostrServiceProvider);
+    final authService = ref.watch(authServiceProvider);
+    final currentUserPubkey = authService.currentPublicKeyHex;
+
+    // Don't show follow button for own profile
+    if (currentUserPubkey == pubkey) {
```

Comment:
note: actually some of the clients do that by default, cause if you follow your self you dont have to add extra logic to see own posts in home screen 😄